src/components/MoneyTemporaryForRefactorRequestConfirmationList.js

```diff
@@ -481,21 +481,29 @@ function MoneyTemporaryForRefactorRequestConfirmationList({
         IOU.setMoneyRequestMerchant(transaction.transactionID, distanceMerchant, true);
     }, [isDistanceRequestWithPendingRoute, hasRoute, distance, unit, rate, currency, translate, toLocaleDigit, isDistanceRequest, transaction]);
 
+    // Auto select the category if there is only one enabled category and it is required
     useEffect(() => {
         const enabledCategories = _.filter(policyCategories, (category) => category.enabled);
-        if (iouCategory || !shouldShowCategories || enabledCategories.length !== 1) {
+        if (iouCategory || !shouldShowCategories || enabledCategories.length !== 1 || !isCategoryRequired) {
```

Comment:
Using `isCategoryRequired` here means we'll get inconsistent behaviour vs tags as `isCategoryRequired` relies on `canUseViolations` being true, whereas the checks for the tags are not reliant on that. Do you think it's better to just use `lodashGet(policy, 'requiresCategory', false)` directly here?

Reply:
@dukenv0307 Ah good point! For multi-level tags I think we need to use the `required` field on each tag. For single-level tags however, we need to use `isTagRequired` as on oldDot this option is only available when multi-level tags are turned off `(People must tag expenses)`:

![image](https://github.com/Expensify/App/assets/27287420/d98a2aa1-e51c-4106-9ac4-6cf6be57aec2)

Regarding my previous comment, I think it probably makes sense to just add `canUseViolations` to the tags to keep things consistent. Tags and Categories don't display the `Required` text unless `canUseViolations` is true so it makes sense for us to tie in with that I think!